### ISSUE TITLE
Left Join with many fail on null

### ISSUE THREAD

**Issue by rkosafo** [rspeele/Rezoom.SQL#24]
Given

```sql
    ( Id int primary key autoincrement
    , Email string(254) unique
    , Name string(64) null
    );

create table Comments
    ( Id int primary key autoincrement
    , AuthorId int references Users(Id)
    , Comment string(512)
    );

create index IX_Comments_AuthorId on Comments
    (AuthorId);

-- initial data
Insert into Users (email, name)
Values ('a@outlook.com', 'Mr. A'), ('b@gmail.com', 'Mr. B');

Insert into Comments (authorId, comment)
Values (1, 'This is cool'), (1, 'This is not cool');
```

The code below fails with `Data is Null. This method or property cannot be called on Null values.`

````sql
module Tests =
  let print x = printf "Output: %A" x

  type LeftJoinWithManyFails = SQL<"""
    Select u.id, many comments (c.id, c.comment)
    From Users u
    left Join Comments c on u.id = c.authorId""">
  let leftJoinWithManyFails () =
    LeftJoinWithManyFails.Command().Execute (new ConnectionContext()) |> print
````

The expectation was to have `comments` as an empty list.  Is this possible?

**Comment by rspeele** [comment]
Agreed, this should work. Will look into it when I get a chance.

**Comment by Risord** [comment]
Has there been any progress related to this?

**Comment by rspeele** [comment]
I have been getting into some new hobbies lately and haven't worked on this much, but this issue is definitely egregious so I have committed a fix in cbd096d .
